### PR TITLE
added days supplied to medication

### PR DESCRIPTION
**Pull request by daco101**
This PR adds a "Days Supplied" attribute to: Medication Order, Dispensed and Discharge. The field is a number indicating the number of days a supply of medication should last.

Pull requests into CQL QDM Patient API require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1690
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass N/A

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint N/A

**Reviewer 1:**

Name: @hossenlopp
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @mayerm94
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: spec/javascripts/datatypes/medication_spec.js.coffee [hunk 1]
```diff
@@ -107,6 +113,7 @@ describe "Medication", ->
       "anatomical_approach":null,
       "codes":{"RxNorm":["1010600"]},
       "cumulativeMedicationDuration":null,
+      "daysSupplied":{"scalar":"5","units":""}
```
Comment (mayerm94) [inline spec/javascripts/datatypes/medication_spec.js.coffee]: Missing comma at the end

File: spec/javascripts/datatypes/medication_spec.js.coffee [hunk 2]
```diff
@@ -358,6 +370,7 @@ describe "Medication", ->
...
```
Comment (mayerm94) [inline spec/javascripts/datatypes/medication_spec.js.coffee]: Missing comma at the end of the line

File: spec/javascripts/datatypes/medication_spec.js.coffee [hunk 3]
```diff
@@ -8,6 +8,7 @@ describe "Medication", ->
...
```
Comment (hossenlopp) [inline spec/javascripts/datatypes/medication_spec.js.coffee]: Missing comma at end of lime.

File: app/assets/javascripts/datatypes/medication.js.coffee [hunk 4]
```diff
@@ -188,6 +189,15 @@ class CQL_QDM.MedicationDischarge extends CQL_QDM.QDMDatatype
   authorDatetime: ->
     @_authorDatetime
 
+  ###
+  @returns {Integer}
+  ###
+  daysSupplied: ->
+    if @_daysSupplied?
```
Comment (hossenlopp) [inline app/assets/javascripts/datatypes/medication.js.coffee]: Should probably check if `@_daySupplied?['scalar']` exists. If the user puts a code in the field, this returns `NaN`. While accurate, probably should return `null`.
